// ng-brazil/esm2015/pispasep/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { pispasep } from './validator';
import * as ɵngcc0 from '@angular/core';
const PISPASE_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => PispasepValidator),
    multi: true
};
export class PispasepValidator {
    validate(c) {
        return pispasep(c);
    }
}
PispasepValidator.ɵfac = function PispasepValidator_Factory(t) { return new (t || PispasepValidator)(); };
PispasepValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: PispasepValidator, selectors: [["", "pispasep", "", "formControlName", ""], ["", "pispasep", "", "formControl", ""], ["", "pispasep", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([PISPASE_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PispasepValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[pispasep][formControlName],[pispasep][formControl],[pispasep][ngModel]',
                providers: [PISPASE_VALIDATOR]
            }]
    }], null, null); })();
