// ng-brazil/esm2015/rg/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { rg } from './validator';
import * as ɵngcc0 from '@angular/core';
const RG_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => RGValidator),
    multi: true
};
export class RGValidator {
    validate(c) {
        return rg(c);
    }
}
RGValidator.ɵfac = function RGValidator_Factory(t) { return new (t || RGValidator)(); };
RGValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RGValidator, selectors: [["", "rg", "", "formControlName", ""], ["", "rg", "", "formControl", ""], ["", "rg", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([RG_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RGValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[rg][formControlName],[rg][formControl],[rg][ngModel]',
                providers: [RG_VALIDATOR]
            }]
    }], null, null); })();
