// ng-brazil/esm2015/placa/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { placa } from './validator';
import * as ɵngcc0 from '@angular/core';
const PLACA_VALIDATOR = {
    provide: NG_VALIDATORS,
    useExisting: forwardRef(() => PLACAValidator),
    multi: true
};
export class PLACAValidator {
    validate(c) {
        return placa(c);
    }
}
PLACAValidator.ɵfac = function PLACAValidator_Factory(t) { return new (t || PLACAValidator)(); };
PLACAValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: PLACAValidator, selectors: [["", "placa", "", "formControlName", ""], ["", "placa", "", "formControl", ""], ["", "placa", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([PLACA_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PLACAValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[placa][formControlName],[placa][formControl],[placa][ngModel]',
                providers: [PLACA_VALIDATOR]
            }]
    }], null, null); })();
