// ng-brazil/esm2015/titulo/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { titulo } from './validator';
import * as ɵngcc0 from '@angular/core';
const TITULO_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => TITULOValidator),
    multi: true
};
export class TITULOValidator {
    validate(c) {
        return titulo(c);
    }
}
TITULOValidator.ɵfac = function TITULOValidator_Factory(t) { return new (t || TITULOValidator)(); };
TITULOValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TITULOValidator, selectors: [["", "titulo", "", "formControlName", ""], ["", "titulo", "", "formControl", ""], ["", "titulo", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([TITULO_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TITULOValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[titulo][formControlName],[titulo][formControl],[titulo][ngModel]',
                providers: [TITULO_VALIDATOR]
            }]
    }], null, null); })();
