// ng-brazil/esm2015/inscricaoestadual/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { inscricaoestadual } from './validator';
import * as ɵngcc0 from '@angular/core';
const INSCRICAOESTADUAL_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => InscricaoEstadualValidator),
    multi: true
};
export class InscricaoEstadualValidator {
    validate(c) {
        return inscricaoestadual('mg')(c);
    }
}
InscricaoEstadualValidator.ɵfac = function InscricaoEstadualValidator_Factory(t) { return new (t || InscricaoEstadualValidator)(); };
InscricaoEstadualValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InscricaoEstadualValidator, selectors: [["", "inscricaoestadual", "", "formControlName", ""], ["", "inscricaoestadual", "", "formControl", ""], ["", "inscricaoestadual", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([INSCRICAOESTADUAL_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InscricaoEstadualValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[inscricaoestadual][formControlName],[inscricaoestadual][formControl],[inscricaoestadual][ngModel]',
                providers: [INSCRICAOESTADUAL_VALIDATOR]
            }]
    }], null, null); })();
