// ng-brazil/esm2015/percentage/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { percentage } from './validator';
import * as ɵngcc0 from '@angular/core';
const PERCENTAGE_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => PERCENTAGEValidator),
    multi: true
};
export class PERCENTAGEValidator {
    validate(c) {
        return percentage(c);
    }
}
PERCENTAGEValidator.ɵfac = function PERCENTAGEValidator_Factory(t) { return new (t || PERCENTAGEValidator)(); };
PERCENTAGEValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: PERCENTAGEValidator, selectors: [["", "percentage", "", "formControlName", ""], ["", "percentage", "", "formControl", ""], ["", "percentage", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([PERCENTAGE_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PERCENTAGEValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[percentage][formControlName],[percentage][formControl],[percentage][ngModel]',
                providers: [PERCENTAGE_VALIDATOR]
            }]
    }], null, null); })();
