// ng-brazil/esm2015/telefone/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { telefone } from './validator';
import * as ɵngcc0 from '@angular/core';
const TELEFONE_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => TelefoneValidator),
    multi: true
};
export class TelefoneValidator {
    validate(c) {
        return telefone(c);
    }
}
TelefoneValidator.ɵfac = function TelefoneValidator_Factory(t) { return new (t || TelefoneValidator)(); };
TelefoneValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TelefoneValidator, selectors: [["", "telefone", "", "formControlName", ""], ["", "telefone", "", "formControl", ""], ["", "telefone", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([TELEFONE_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TelefoneValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[telefone][formControlName],[telefone][formControl],[telefone][ngModel]',
                providers: [TELEFONE_VALIDATOR]
            }]
    }], null, null); })();
