// ng-brazil/esm2015/celular/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { celular } from './validator';
import * as ɵngcc0 from '@angular/core';
const CELULAR_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => CelularValidator),
    multi: true
};
export class CelularValidator {
    validate(c) {
        return celular(c);
    }
}
CelularValidator.ɵfac = function CelularValidator_Factory(t) { return new (t || CelularValidator)(); };
CelularValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CelularValidator, selectors: [["", "celular", "", "formControlName", ""], ["", "celular", "", "formControl", ""], ["", "celular", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CELULAR_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CelularValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[celular][formControlName],[celular][formControl],[celular][ngModel]',
                providers: [CELULAR_VALIDATOR]
            }]
    }], null, null); })();
