// ng-brazil/esm2015/cnpj/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { cnpj } from './validator';
import * as ɵngcc0 from '@angular/core';
const CNPJ_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => CNPJValidator),
    multi: true
};
export class CNPJValidator {
    validate(c) {
        return cnpj(c);
    }
}
CNPJValidator.ɵfac = function CNPJValidator_Factory(t) { return new (t || CNPJValidator)(); };
CNPJValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CNPJValidator, selectors: [["", "cnpj", "", "formControlName", ""], ["", "cnpj", "", "formControl", ""], ["", "cnpj", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CNPJ_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CNPJValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[cnpj][formControlName],[cnpj][formControl],[cnpj][ngModel]',
                providers: [CNPJ_VALIDATOR]
            }]
    }], null, null); })();
