// ng-brazil/esm2015/time/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { time } from './validator';
import * as ɵngcc0 from '@angular/core';
const TIME_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => TIMEValidator),
    multi: true
};
export class TIMEValidator {
    validate(c) {
        return time(c);
    }
}
TIMEValidator.ɵfac = function TIMEValidator_Factory(t) { return new (t || TIMEValidator)(); };
TIMEValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TIMEValidator, selectors: [["", "time", "", "formControlName", ""], ["", "time", "", "formControl", ""], ["", "time", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([TIME_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TIMEValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[time][formControlName],[time][formControl],[time][ngModel]',
                providers: [TIME_VALIDATOR]
            }]
    }], null, null); })();
