// ng-brazil/esm2015/cep/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { cep } from './validator';
import * as ɵngcc0 from '@angular/core';
const CEP_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => CEPValidator),
    multi: true
};
export class CEPValidator {
    validate(c) {
        return cep(c);
    }
}
CEPValidator.ɵfac = function CEPValidator_Factory(t) { return new (t || CEPValidator)(); };
CEPValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CEPValidator, selectors: [["", "cep", "", "formControlName", ""], ["", "cep", "", "formControl", ""], ["", "cep", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CEP_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CEPValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[cep][formControlName],[cep][formControl],[cep][ngModel]',
                providers: [CEP_VALIDATOR]
            }]
    }], null, null); })();
