// ng-brazil/esm2015/renavam/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { renavam } from './validator';
import * as ɵngcc0 from '@angular/core';
const RENAVAM_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => RenavamValidator),
    multi: true
};
export class RenavamValidator {
    validate(c) {
        return renavam(c);
    }
}
RenavamValidator.ɵfac = function RenavamValidator_Factory(t) { return new (t || RenavamValidator)(); };
RenavamValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RenavamValidator, selectors: [["", "renavam", "", "formControlName", ""], ["", "renavam", "", "formControl", ""], ["", "renavam", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([RENAVAM_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RenavamValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[renavam][formControlName],[renavam][formControl],[renavam][ngModel]',
                providers: [RENAVAM_VALIDATOR]
            }]
    }], null, null); })();
