// ng-brazil/bundles/ng-brazil.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/platform-browser'),exports, require('@angular/core'), require('@angular/forms'), require('js-brasil'), require('@angular/platform-browser'), require('text-mask-core/dist/textMaskCore')) :
    typeof define === 'function' && define.amd ? define('ng-brazil', ['@angular/core','@angular/platform-browser','exports', '@angular/core', '@angular/forms', 'js-brasil', '@angular/platform-browser', 'text-mask-core/dist/textMaskCore'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.ng.core,global.ng.platformBrowser,global['ng-brazil'] = {}, global.ng.core, global.ng.forms, global.jsBrasil, global.ng.platformBrowser, global.textMaskCore));
}(this, (function (ɵngcc0,ɵngcc1,exports, core, forms, jsBrasil, platformBrowser, textMaskCore) { 'use strict';

    var cpf = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.cpf(v) ? null : { cpf: true };
    };

    var CPF_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return CPFValidator; }),
        multi: true
    };
    var CPFValidator = /** @class */ (function () {
        function CPFValidator() {
        }
        CPFValidator.prototype.validate = function (c) {
            return cpf(c);
        };
CPFValidator.ɵfac = function CPFValidator_Factory(t) { return new (t || CPFValidator)(); };
CPFValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CPFValidator, selectors: [["", "cpf", "", "formControlName", ""], ["", "cpf", "", "formControl", ""], ["", "cpf", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CPF_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CPFValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[cpf][formControlName],[cpf][formControl],[cpf][ngModel]',
                providers: [CPF_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return CPFValidator;
    }());

    var CPFPipe = /** @class */ (function () {
        function CPFPipe() {
        }
        CPFPipe.prototype.transform = function (cpfValue) {
            return jsBrasil.maskBr.cpf(cpfValue);
        };
CPFPipe.ɵfac = function CPFPipe_Factory(t) { return new (t || CPFPipe)(); };
CPFPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "cpf", type: CPFPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CPFPipe, [{
        type: core.Pipe,
        args: [{
                name: 'cpf'
            }]
    }], function () { return []; }, null); })();
        return CPFPipe;
    }());

    var cnpj = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.cnpj(v) ? null : { cnpj: true };
    };

    var CNPJ_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return CNPJValidator; }),
        multi: true
    };
    var CNPJValidator = /** @class */ (function () {
        function CNPJValidator() {
        }
        CNPJValidator.prototype.validate = function (c) {
            return cnpj(c);
        };
CNPJValidator.ɵfac = function CNPJValidator_Factory(t) { return new (t || CNPJValidator)(); };
CNPJValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CNPJValidator, selectors: [["", "cnpj", "", "formControlName", ""], ["", "cnpj", "", "formControl", ""], ["", "cnpj", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CNPJ_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CNPJValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[cnpj][formControlName],[cnpj][formControl],[cnpj][ngModel]',
                providers: [CNPJ_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return CNPJValidator;
    }());

    var CNPJPipe = /** @class */ (function () {
        function CNPJPipe() {
        }
        CNPJPipe.prototype.transform = function (cnpjValue) {
            return jsBrasil.maskBr.cnpj(cnpjValue);
        };
CNPJPipe.ɵfac = function CNPJPipe_Factory(t) { return new (t || CNPJPipe)(); };
CNPJPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "cnpj", type: CNPJPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CNPJPipe, [{
        type: core.Pipe,
        args: [{
                name: 'cnpj'
            }]
    }], function () { return []; }, null); })();
        return CNPJPipe;
    }());

    var telefone = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.telefone(v) ? null : { telefone: true };
    };

    var TELEFONE_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return TelefoneValidator; }),
        multi: true
    };
    var TelefoneValidator = /** @class */ (function () {
        function TelefoneValidator() {
        }
        TelefoneValidator.prototype.validate = function (c) {
            return telefone(c);
        };
TelefoneValidator.ɵfac = function TelefoneValidator_Factory(t) { return new (t || TelefoneValidator)(); };
TelefoneValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TelefoneValidator, selectors: [["", "telefone", "", "formControlName", ""], ["", "telefone", "", "formControl", ""], ["", "telefone", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([TELEFONE_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TelefoneValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[telefone][formControlName],[telefone][formControl],[telefone][ngModel]',
                providers: [TELEFONE_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return TelefoneValidator;
    }());

    var TelefonePipe = /** @class */ (function () {
        function TelefonePipe() {
        }
        TelefonePipe.prototype.transform = function (telefoneValue) {
            return jsBrasil.maskBr.telefone(telefoneValue);
        };
TelefonePipe.ɵfac = function TelefonePipe_Factory(t) { return new (t || TelefonePipe)(); };
TelefonePipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "telefone", type: TelefonePipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TelefonePipe, [{
        type: core.Pipe,
        args: [{
                name: 'telefone'
            }]
    }], function () { return []; }, null); })();
        return TelefonePipe;
    }());

    var celular = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.celular(v) ? null : { celular: true };
    };

    var CELULAR_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return CelularValidator; }),
        multi: true
    };
    var CelularValidator = /** @class */ (function () {
        function CelularValidator() {
        }
        CelularValidator.prototype.validate = function (c) {
            return celular(c);
        };
CelularValidator.ɵfac = function CelularValidator_Factory(t) { return new (t || CelularValidator)(); };
CelularValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CelularValidator, selectors: [["", "celular", "", "formControlName", ""], ["", "celular", "", "formControl", ""], ["", "celular", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CELULAR_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CelularValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[celular][formControlName],[celular][formControl],[celular][ngModel]',
                providers: [CELULAR_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return CelularValidator;
    }());

    var CelularPipe = /** @class */ (function () {
        function CelularPipe() {
        }
        CelularPipe.prototype.transform = function (celularValue) {
            return jsBrasil.maskBr.celular(celularValue);
        };
CelularPipe.ɵfac = function CelularPipe_Factory(t) { return new (t || CelularPipe)(); };
CelularPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "celular", type: CelularPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CelularPipe, [{
        type: core.Pipe,
        args: [{
                name: 'celular'
            }]
    }], function () { return []; }, null); })();
        return CelularPipe;
    }());

    var inscricaoestadual = function (estado) {
        return function (control) {
            if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
                return null;
            }
            var v = control.value;
            return jsBrasil.validateBr.inscricaoestadual(v, estado) ? null : { inscricaoestadual: true };
        };
    };

    var INSCRICAOESTADUAL_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return InscricaoEstadualValidator; }),
        multi: true
    };
    var InscricaoEstadualValidator = /** @class */ (function () {
        function InscricaoEstadualValidator() {
        }
        InscricaoEstadualValidator.prototype.validate = function (c) {
            return inscricaoestadual('mg')(c);
        };
InscricaoEstadualValidator.ɵfac = function InscricaoEstadualValidator_Factory(t) { return new (t || InscricaoEstadualValidator)(); };
InscricaoEstadualValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InscricaoEstadualValidator, selectors: [["", "inscricaoestadual", "", "formControlName", ""], ["", "inscricaoestadual", "", "formControl", ""], ["", "inscricaoestadual", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([INSCRICAOESTADUAL_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InscricaoEstadualValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[inscricaoestadual][formControlName],[inscricaoestadual][formControl],[inscricaoestadual][ngModel]',
                providers: [INSCRICAOESTADUAL_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return InscricaoEstadualValidator;
    }());

    var InscricaoEstadualPipe = /** @class */ (function () {
        function InscricaoEstadualPipe() {
        }
        InscricaoEstadualPipe.prototype.transform = function (inscricaoestadualValue, estado) {
            return jsBrasil.maskBr.inscricaoestadual(inscricaoestadualValue, estado);
        };
InscricaoEstadualPipe.ɵfac = function InscricaoEstadualPipe_Factory(t) { return new (t || InscricaoEstadualPipe)(); };
InscricaoEstadualPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "inscricaoestadual", type: InscricaoEstadualPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InscricaoEstadualPipe, [{
        type: core.Pipe,
        args: [{
                name: 'inscricaoestadual'
            }]
    }], function () { return []; }, null); })();
        return InscricaoEstadualPipe;
    }());

    var cep = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.cep(v) ? null : { cep: true };
    };

    var CEP_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return CEPValidator; }),
        multi: true
    };
    var CEPValidator = /** @class */ (function () {
        function CEPValidator() {
        }
        CEPValidator.prototype.validate = function (c) {
            return cep(c);
        };
CEPValidator.ɵfac = function CEPValidator_Factory(t) { return new (t || CEPValidator)(); };
CEPValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CEPValidator, selectors: [["", "cep", "", "formControlName", ""], ["", "cep", "", "formControl", ""], ["", "cep", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CEP_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CEPValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[cep][formControlName],[cep][formControl],[cep][ngModel]',
                providers: [CEP_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return CEPValidator;
    }());

    var CEPPipe = /** @class */ (function () {
        function CEPPipe() {
        }
        CEPPipe.prototype.transform = function (cepValue) {
            return jsBrasil.maskBr.cep(cepValue);
        };
CEPPipe.ɵfac = function CEPPipe_Factory(t) { return new (t || CEPPipe)(); };
CEPPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "cep", type: CEPPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CEPPipe, [{
        type: core.Pipe,
        args: [{
                name: 'cep'
            }]
    }], function () { return []; }, null); })();
        return CEPPipe;
    }());

    var currency = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.currency(v) ? null : { currency: true };
    };

    var CURRENCY_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return CURRENCYValidator; }),
        multi: true
    };
    var CURRENCYValidator = /** @class */ (function () {
        function CURRENCYValidator() {
        }
        CURRENCYValidator.prototype.validate = function (c) {
            return currency(c);
        };
CURRENCYValidator.ɵfac = function CURRENCYValidator_Factory(t) { return new (t || CURRENCYValidator)(); };
CURRENCYValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CURRENCYValidator, selectors: [["", "currency", "", "formControlName", ""], ["", "currency", "", "formControl", ""], ["", "currency", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CURRENCY_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CURRENCYValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[currency][formControlName],[currency][formControl],[currency][ngModel]',
                providers: [CURRENCY_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return CURRENCYValidator;
    }());

    var CURRENCYPipe = /** @class */ (function () {
        function CURRENCYPipe() {
        }
        CURRENCYPipe.prototype.transform = function (currencyValue) {
            return jsBrasil.maskBr.currency(currencyValue);
        };
CURRENCYPipe.ɵfac = function CURRENCYPipe_Factory(t) { return new (t || CURRENCYPipe)(); };
CURRENCYPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "currencyBrazil", type: CURRENCYPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CURRENCYPipe, [{
        type: core.Pipe,
        args: [{
                name: 'currencyBrazil'
            }]
    }], function () { return []; }, null); })();
        return CURRENCYPipe;
    }());

    var number = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.number(v) ? null : { number: true };
    };

    var NUMBER_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return NUMBERValidator; }),
        multi: true
    };
    var NUMBERValidator = /** @class */ (function () {
        function NUMBERValidator() {
        }
        NUMBERValidator.prototype.validate = function (c) {
            return number(c);
        };
NUMBERValidator.ɵfac = function NUMBERValidator_Factory(t) { return new (t || NUMBERValidator)(); };
NUMBERValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NUMBERValidator, selectors: [["", "number", "", "formControlName", ""], ["", "number", "", "formControl", ""], ["", "number", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([NUMBER_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NUMBERValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[number][formControlName],[number][formControl],[number][ngModel]',
                providers: [NUMBER_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return NUMBERValidator;
    }());

    var NUMBERPipe = /** @class */ (function () {
        function NUMBERPipe() {
        }
        NUMBERPipe.prototype.transform = function (numberValue) {
            return jsBrasil.maskBr.number(numberValue);
        };
NUMBERPipe.ɵfac = function NUMBERPipe_Factory(t) { return new (t || NUMBERPipe)(); };
NUMBERPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "numberBrazil", type: NUMBERPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NUMBERPipe, [{
        type: core.Pipe,
        args: [{
                name: 'numberBrazil'
            }]
    }], function () { return []; }, null); })();
        return NUMBERPipe;
    }());

    var placa = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.placa(v) ? null : { placa: true };
    };

    var PLACA_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        useExisting: core.forwardRef(function () { return PLACAValidator; }),
        multi: true
    };
    var PLACAValidator = /** @class */ (function () {
        function PLACAValidator() {
        }
        PLACAValidator.prototype.validate = function (c) {
            return placa(c);
        };
PLACAValidator.ɵfac = function PLACAValidator_Factory(t) { return new (t || PLACAValidator)(); };
PLACAValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: PLACAValidator, selectors: [["", "placa", "", "formControlName", ""], ["", "placa", "", "formControl", ""], ["", "placa", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([PLACA_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PLACAValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[placa][formControlName],[placa][formControl],[placa][ngModel]',
                providers: [PLACA_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return PLACAValidator;
    }());

    var PLACAPipe = /** @class */ (function () {
        function PLACAPipe() {
        }
        PLACAPipe.prototype.transform = function (placaValue) {
            return jsBrasil.maskBr.placa(placaValue);
        };
PLACAPipe.ɵfac = function PLACAPipe_Factory(t) { return new (t || PLACAPipe)(); };
PLACAPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "placa", type: PLACAPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PLACAPipe, [{
        type: core.Pipe,
        args: [{
                name: 'placa'
            }]
    }], function () { return []; }, null); })();
        return PLACAPipe;
    }());

    var percentage = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.porcentagem(v) ? null : { percentage: true };
    };

    var PERCENTAGE_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return PERCENTAGEValidator; }),
        multi: true
    };
    var PERCENTAGEValidator = /** @class */ (function () {
        function PERCENTAGEValidator() {
        }
        PERCENTAGEValidator.prototype.validate = function (c) {
            return percentage(c);
        };
PERCENTAGEValidator.ɵfac = function PERCENTAGEValidator_Factory(t) { return new (t || PERCENTAGEValidator)(); };
PERCENTAGEValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: PERCENTAGEValidator, selectors: [["", "percentage", "", "formControlName", ""], ["", "percentage", "", "formControl", ""], ["", "percentage", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([PERCENTAGE_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PERCENTAGEValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[percentage][formControlName],[percentage][formControl],[percentage][ngModel]',
                providers: [PERCENTAGE_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return PERCENTAGEValidator;
    }());

    var PERCENTAGEPipe = /** @class */ (function () {
        function PERCENTAGEPipe() {
        }
        PERCENTAGEPipe.prototype.transform = function (percentageValue) {
            return jsBrasil.maskBr.porcentagem(percentageValue);
        };
PERCENTAGEPipe.ɵfac = function PERCENTAGEPipe_Factory(t) { return new (t || PERCENTAGEPipe)(); };
PERCENTAGEPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "percentage", type: PERCENTAGEPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PERCENTAGEPipe, [{
        type: core.Pipe,
        args: [{
                name: 'percentage'
            }]
    }], function () { return []; }, null); })();
        return PERCENTAGEPipe;
    }());

    var renavam = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.renavam(v) ? null : { renavam: true };
    };

    var RENAVAM_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return RenavamValidator; }),
        multi: true
    };
    var RenavamValidator = /** @class */ (function () {
        function RenavamValidator() {
        }
        RenavamValidator.prototype.validate = function (c) {
            return renavam(c);
        };
RenavamValidator.ɵfac = function RenavamValidator_Factory(t) { return new (t || RenavamValidator)(); };
RenavamValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RenavamValidator, selectors: [["", "renavam", "", "formControlName", ""], ["", "renavam", "", "formControl", ""], ["", "renavam", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([RENAVAM_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RenavamValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[renavam][formControlName],[renavam][formControl],[renavam][ngModel]',
                providers: [RENAVAM_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return RenavamValidator;
    }());

    var RenavamPipe = /** @class */ (function () {
        function RenavamPipe() {
        }
        RenavamPipe.prototype.transform = function (renavamValue) {
            return jsBrasil.maskBr.renavam(renavamValue);
        };
RenavamPipe.ɵfac = function RenavamPipe_Factory(t) { return new (t || RenavamPipe)(); };
RenavamPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "renavam", type: RenavamPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RenavamPipe, [{
        type: core.Pipe,
        args: [{
                name: 'renavam'
            }]
    }], function () { return []; }, null); })();
        return RenavamPipe;
    }());

    var pispasep = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.pispasep(v) ? null : { pispasep: true };
    };

    var PISPASE_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return PispasepValidator; }),
        multi: true
    };
    var PispasepValidator = /** @class */ (function () {
        function PispasepValidator() {
        }
        PispasepValidator.prototype.validate = function (c) {
            return pispasep(c);
        };
PispasepValidator.ɵfac = function PispasepValidator_Factory(t) { return new (t || PispasepValidator)(); };
PispasepValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: PispasepValidator, selectors: [["", "pispasep", "", "formControlName", ""], ["", "pispasep", "", "formControl", ""], ["", "pispasep", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([PISPASE_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PispasepValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[pispasep][formControlName],[pispasep][formControl],[pispasep][ngModel]',
                providers: [PISPASE_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return PispasepValidator;
    }());

    var PispasepPipe = /** @class */ (function () {
        function PispasepPipe() {
        }
        PispasepPipe.prototype.transform = function (pispasepValue) {
            return jsBrasil.maskBr.pispasep(pispasepValue);
        };
PispasepPipe.ɵfac = function PispasepPipe_Factory(t) { return new (t || PispasepPipe)(); };
PispasepPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "pispasep", type: PispasepPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PispasepPipe, [{
        type: core.Pipe,
        args: [{
                name: 'pispasep'
            }]
    }], function () { return []; }, null); })();
        return PispasepPipe;
    }());

    var rg = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.rg(v) ? null : { rg: true };
    };

    var RG_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return RGValidator; }),
        multi: true
    };
    var RGValidator = /** @class */ (function () {
        function RGValidator() {
        }
        RGValidator.prototype.validate = function (c) {
            return rg(c);
        };
RGValidator.ɵfac = function RGValidator_Factory(t) { return new (t || RGValidator)(); };
RGValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RGValidator, selectors: [["", "rg", "", "formControlName", ""], ["", "rg", "", "formControl", ""], ["", "rg", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([RG_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RGValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[rg][formControlName],[rg][formControl],[rg][ngModel]',
                providers: [RG_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return RGValidator;
    }());

    var RGPipe = /** @class */ (function () {
        function RGPipe() {
        }
        RGPipe.prototype.transform = function (rgValue) {
            return jsBrasil.maskBr.rg(rgValue);
        };
RGPipe.ɵfac = function RGPipe_Factory(t) { return new (t || RGPipe)(); };
RGPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "rg", type: RGPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RGPipe, [{
        type: core.Pipe,
        args: [{
                name: 'rg'
            }]
    }], function () { return []; }, null); })();
        return RGPipe;
    }());

    var time = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.time(v) ? null : { time: true };
    };

    var TIME_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return TIMEValidator; }),
        multi: true
    };
    var TIMEValidator = /** @class */ (function () {
        function TIMEValidator() {
        }
        TIMEValidator.prototype.validate = function (c) {
            return time(c);
        };
TIMEValidator.ɵfac = function TIMEValidator_Factory(t) { return new (t || TIMEValidator)(); };
TIMEValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TIMEValidator, selectors: [["", "time", "", "formControlName", ""], ["", "time", "", "formControl", ""], ["", "time", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([TIME_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TIMEValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[time][formControlName],[time][formControl],[time][ngModel]',
                providers: [TIME_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return TIMEValidator;
    }());

    var TIMEPipe = /** @class */ (function () {
        function TIMEPipe() {
        }
        TIMEPipe.prototype.transform = function (timeValue) {
            return jsBrasil.maskBr.time(timeValue);
        };
TIMEPipe.ɵfac = function TIMEPipe_Factory(t) { return new (t || TIMEPipe)(); };
TIMEPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "time", type: TIMEPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TIMEPipe, [{
        type: core.Pipe,
        args: [{
                name: 'time'
            }]
    }], function () { return []; }, null); })();
        return TIMEPipe;
    }());

    var titulo = function (control) {
        if (jsBrasil.utilsBr.isPresent(forms.Validators.required(control))) {
            return null;
        }
        var v = control.value;
        return jsBrasil.validateBr.titulo(v) ? null : { titulo: true };
    };

    var TITULO_VALIDATOR = {
        provide: forms.NG_VALIDATORS,
        // tslint:disable-next-line: no-use-before-declare
        useExisting: core.forwardRef(function () { return TITULOValidator; }),
        multi: true
    };
    var TITULOValidator = /** @class */ (function () {
        function TITULOValidator() {
        }
        TITULOValidator.prototype.validate = function (c) {
            return titulo(c);
        };
TITULOValidator.ɵfac = function TITULOValidator_Factory(t) { return new (t || TITULOValidator)(); };
TITULOValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TITULOValidator, selectors: [["", "titulo", "", "formControlName", ""], ["", "titulo", "", "formControl", ""], ["", "titulo", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([TITULO_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TITULOValidator, [{
        type: core.Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[titulo][formControlName],[titulo][formControl],[titulo][ngModel]',
                providers: [TITULO_VALIDATOR]
            }]
    }], function () { return []; }, null); })();
        return TITULOValidator;
    }());

    var TITULOPipe = /** @class */ (function () {
        function TITULOPipe() {
        }
        TITULOPipe.prototype.transform = function (tituloValue) {
            return jsBrasil.maskBr.titulo(tituloValue);
        };
TITULOPipe.ɵfac = function TITULOPipe_Factory(t) { return new (t || TITULOPipe)(); };
TITULOPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "titulo", type: TITULOPipe, pure: true });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TITULOPipe, [{
        type: core.Pipe,
        args: [{
                name: 'titulo'
            }]
    }], function () { return []; }, null); })();
        return TITULOPipe;
    }());

    var TextMaskConfig = /** @class */ (function () {
        function TextMaskConfig() {
        }
        return TextMaskConfig;
    }());
    var MASKEDINPUT_VALUE_ACCESSOR = {
        provide: forms.NG_VALUE_ACCESSOR,
        useExisting: core.forwardRef(function () { return MaskedInputDirective; }),
        multi: true
    };
    /**
     * We must check whether the agent is Android because composition events
     * behave differently between iOS and Android.
     */
    function _isAndroid() {
        var userAgent = platformBrowser.ɵgetDOM() ? platformBrowser.ɵgetDOM().getUserAgent() : '';
        return /android (\d+)/.test(userAgent.toLowerCase());
    }
    var MaskedInputDirective = /** @class */ (function () {
        function MaskedInputDirective(rendererFactory, _elementRef, _compositionMode) {
            this._elementRef = _elementRef;
            this._compositionMode = _compositionMode;
            this.textMaskConfig = {
                mask: [],
                guide: true,
                placeholderChar: '_',
                pipe: undefined,
                keepCharPositions: false,
            };
            this.onChange = function (_) { };
            this.onTouched = function () { };
            /** Whether the user is creating a composition string (IME events). */
            this._composing = false;
            this._renderer = rendererFactory.createRenderer(null, null);
            if (this._compositionMode == null) {
                this._compositionMode = !_isAndroid();
            }
        }
        MaskedInputDirective.prototype.ngOnChanges = function (changes) {
            this._setupMask(true);
            if (this.textMaskInputElement !== undefined) {
                this.textMaskInputElement.update(this.inputElement.value);
            }
        };
        MaskedInputDirective.prototype.writeValue = function (value) {
            this._setupMask();
            // set the initial value for cases where the mask is disabled
            var normalizedValue = value == null ? '' : value;
            this._renderer.setProperty(this.inputElement, 'value', normalizedValue);
            if (this.textMaskInputElement !== undefined) {
                this.textMaskInputElement.update(value);
            }
        };
        MaskedInputDirective.prototype.registerOnChange = function (fn) { this.onChange = fn; };
        MaskedInputDirective.prototype.registerOnTouched = function (fn) { this.onTouched = fn; };
        MaskedInputDirective.prototype.setDisabledState = function (isDisabled) {
            this._renderer.setProperty(this._elementRef.nativeElement, 'disabled', isDisabled);
        };
        MaskedInputDirective.prototype._handleInput = function (value) {
            if (!this._compositionMode || (this._compositionMode && !this._composing)) {
                this._setupMask();
                if (this.textMaskInputElement !== undefined) {
                    this.textMaskInputElement.update(value);
                    // get the updated value
                    value = this.inputElement.value;
                    this.onChange(value);
                }
            }
        };
        MaskedInputDirective.prototype._setupMask = function (create) {
            if (create === void 0) { create = false; }
            if (!this.inputElement) {
                if (this._elementRef.nativeElement.tagName.toUpperCase() === 'INPUT') {
                    // `textMask` directive is used directly on an input element
                    this.inputElement = this._elementRef.nativeElement;
                }
                else {
                    // `textMask` directive is used on an abstracted input element, `md-input-container`, etc
                    this.inputElement = this._elementRef.nativeElement.getElementsByTagName('INPUT')[0];
                }
            }
            if (this.inputElement && create) {
                this.textMaskInputElement = textMaskCore.createTextMaskInputElement(Object.assign({ inputElement: this.inputElement }, this.textMaskConfig));
            }
        };
        MaskedInputDirective.prototype._compositionStart = function () { this._composing = true; };
        MaskedInputDirective.prototype._compositionEnd = function (value) {
            this._composing = false;
            this._compositionMode && this._handleInput(value);
        };
MaskedInputDirective.ɵfac = function MaskedInputDirective_Factory(t) { return new (t || MaskedInputDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.RendererFactory2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(forms.COMPOSITION_BUFFER_MODE, 8)); };
MaskedInputDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: MaskedInputDirective, selectors: [["", "textMask", ""]], hostBindings: function MaskedInputDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("input", function MaskedInputDirective_input_HostBindingHandler($event) { return ctx._handleInput($event.target.value); })("blur", function MaskedInputDirective_blur_HostBindingHandler() { return ctx.onTouched(); })("compositionstart", function MaskedInputDirective_compositionstart_HostBindingHandler() { return ctx._compositionStart(); })("compositionend", function MaskedInputDirective_compositionend_HostBindingHandler($event) { return ctx._compositionEnd($event.target.value); });
    } }, inputs: { textMaskConfig: ["textMask", "textMaskConfig"] }, exportAs: ["textMask"], features: [ɵngcc0.ɵɵProvidersFeature([MASKEDINPUT_VALUE_ACCESSOR]), ɵngcc0.ɵɵNgOnChangesFeature] });
MaskedInputDirective.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: MaskedInputDirective, factory: function (t) { return MaskedInputDirective.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaskedInputDirective, [{
        type: core.Injectable
    }, {
        type: core.Directive,
        args: [{
                host: {
                    '(input)': '_handleInput($event.target.value)',
                    '(blur)': 'onTouched()',
                    '(compositionstart)': '_compositionStart()',
                    '(compositionend)': '_compositionEnd($event.target.value)'
                },
                selector: '[textMask]',
                exportAs: 'textMask',
                providers: [MASKEDINPUT_VALUE_ACCESSOR]
            }]
    }], function () { return [{ type: ɵngcc0.RendererFactory2 }, { type: ɵngcc0.ElementRef }, { type: Boolean, decorators: [{
                type: core.Optional
            }, {
                type: core.Inject,
                args: [forms.COMPOSITION_BUFFER_MODE]
            }] }]; }, { textMaskConfig: [{
            type: core.Input,
            args: ['textMask']
        }] }); })();
        return MaskedInputDirective;
    }());
    MaskedInputDirective.ctorParameters = function () { return [
        { type: core.RendererFactory2 },
        { type: core.ElementRef },
        { type: Boolean, decorators: [{ type: core.Optional }, { type: core.Inject, args: [forms.COMPOSITION_BUFFER_MODE,] }] }
    ]; };
    MaskedInputDirective.propDecorators = {
        textMaskConfig: [{ type: core.Input, args: ['textMask',] }]
    };
    var TextMaskModule = /** @class */ (function () {
        function TextMaskModule() {
        }
TextMaskModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TextMaskModule });
TextMaskModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TextMaskModule_Factory(t) { return new (t || TextMaskModule)(); }, imports: [[
            platformBrowser.BrowserModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TextMaskModule, { declarations: [MaskedInputDirective], imports: [ɵngcc1.BrowserModule], exports: [MaskedInputDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TextMaskModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    platformBrowser.BrowserModule
                ],
                declarations: [MaskedInputDirective],
                exports: [MaskedInputDirective]
            }]
    }], function () { return []; }, null); })();
        return TextMaskModule;
    }());

    var TextMask = {
        TextMaskModule: TextMaskModule
    };
    var NgBrazilValidators = {
        cpf: cpf,
        cnpj: cnpj,
        celular: celular,
        cep: cep,
        currency: currency,
        inscricaoestadual: inscricaoestadual,
        number: number,
        pispasep: pispasep,
        placa: placa,
        percentage: percentage,
        renavam: renavam,
        rg: rg,
        telefone: telefone,
        time: time,
        titulo: titulo
    };
    var MASKS = jsBrasil.utilsBr.MASKS;
    var CustomDirectives = [
        CelularValidator,
        CelularPipe,
        CEPValidator,
        CEPPipe,
        CPFValidator,
        CPFPipe,
        CNPJValidator,
        CNPJPipe,
        CURRENCYValidator,
        CURRENCYPipe,
        InscricaoEstadualValidator,
        InscricaoEstadualPipe,
        NUMBERValidator,
        NUMBERPipe,
        PERCENTAGEValidator,
        PERCENTAGEPipe,
        PispasepValidator,
        PispasepPipe,
        PLACAValidator,
        PLACAPipe,
        RenavamValidator,
        RenavamPipe,
        RGValidator,
        RGPipe,
        TelefoneValidator,
        TelefonePipe,
        TIMEValidator,
        TIMEPipe,
        TITULOValidator,
        TITULOPipe
    ];
    var NgBrDirectives = {
        CelularValidator: CelularValidator,
        CelularPipe: CelularPipe,
        CEPValidator: CEPValidator,
        CEPPipe: CEPPipe,
        CNPJValidator: CNPJValidator,
        CNPJPipe: CNPJPipe,
        CPFValidator: CPFValidator,
        CPFPipe: CPFPipe,
        CURRENCYValidator: CURRENCYValidator,
        CURRENCYPipe: CURRENCYPipe,
        InscricaoEstadualValidator: InscricaoEstadualValidator,
        InscricaoEstadualPipe: InscricaoEstadualPipe,
        NUMBERValidator: NUMBERValidator,
        NUMBERPipe: NUMBERPipe,
        PERCENTAGEValidator: PERCENTAGEValidator,
        PERCENTAGEPipe: PERCENTAGEPipe,
        PispasepValidator: PispasepValidator,
        PispasepPipe: PispasepPipe,
        PLACAValidator: PLACAValidator,
        PLACAPipe: PLACAPipe,
        RGValidator: RGValidator,
        RGPipe: RGPipe,
        RenavamValidator: RenavamValidator,
        RenavamPipe: RenavamPipe,
        TelefoneValidator: TelefoneValidator,
        TelefonePipe: TelefonePipe,
        TIMEValidator: TIMEValidator,
        TIMEPipe: TIMEPipe,
        TITULOValidator: TITULOValidator,
        TITULOPipe: TITULOPipe
    };
    var NgBrazil = /** @class */ (function () {
        function NgBrazil() {
        }
        NgBrazil.forRoot = function () {
            return {
                ngModule: NgBrazil
            };
        };
NgBrazil.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgBrazil });
NgBrazil.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgBrazil_Factory(t) { return new (t || NgBrazil)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgBrazil, { declarations: [CelularValidator,
        CelularPipe,
        CEPValidator,
        CEPPipe,
        CPFValidator,
        CPFPipe,
        CNPJValidator,
        CNPJPipe,
        CURRENCYValidator,
        CURRENCYPipe,
        InscricaoEstadualValidator,
        InscricaoEstadualPipe,
        NUMBERValidator,
        NUMBERPipe,
        PERCENTAGEValidator,
        PERCENTAGEPipe,
        PispasepValidator,
        PispasepPipe,
        PLACAValidator,
        PLACAPipe,
        RenavamValidator,
        RenavamPipe,
        RGValidator,
        RGPipe,
        TelefoneValidator,
        TelefonePipe,
        TIMEValidator,
        TIMEPipe,
        TITULOValidator,
        TITULOPipe], exports: [CelularValidator,
        CelularPipe,
        CEPValidator,
        CEPPipe,
        CPFValidator,
        CPFPipe,
        CNPJValidator,
        CNPJPipe,
        CURRENCYValidator,
        CURRENCYPipe,
        InscricaoEstadualValidator,
        InscricaoEstadualPipe,
        NUMBERValidator,
        NUMBERPipe,
        PERCENTAGEValidator,
        PERCENTAGEPipe,
        PispasepValidator,
        PispasepPipe,
        PLACAValidator,
        PLACAPipe,
        RenavamValidator,
        RenavamPipe,
        RGValidator,
        RGPipe,
        TelefoneValidator,
        TelefonePipe,
        TIMEValidator,
        TIMEPipe,
        TITULOValidator,
        TITULOPipe] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgBrazil, [{
        type: core.NgModule,
        args: [{
                imports: [],
                declarations: [
                    CustomDirectives
                ],
                exports: [
                    CustomDirectives
                ]
            }]
    }], function () { return []; }, null); })();
        return NgBrazil;
    }());

    /**
     * Generated bundle index. Do not edit.
     */

    exports.CustomDirectives = CustomDirectives;
    exports.MASKS = MASKS;
    exports.NgBrDirectives = NgBrDirectives;
    exports.NgBrazil = NgBrazil;
    exports.NgBrazilValidators = NgBrazilValidators;
    exports.TextMask = TextMask;
    exports.ɵa = MASKEDINPUT_VALUE_ACCESSOR;
    exports.ɵb = MaskedInputDirective;
    exports.ɵba = CURRENCYValidator;
    exports.ɵbb = CURRENCYPipe;
    exports.ɵbc = InscricaoEstadualValidator;
    exports.ɵbd = InscricaoEstadualPipe;
    exports.ɵbe = NUMBERValidator;
    exports.ɵbf = NUMBERPipe;
    exports.ɵbg = PERCENTAGEValidator;
    exports.ɵbh = PERCENTAGEPipe;
    exports.ɵbi = PispasepValidator;
    exports.ɵbj = PispasepPipe;
    exports.ɵbk = PLACAValidator;
    exports.ɵbl = PLACAPipe;
    exports.ɵbm = RenavamValidator;
    exports.ɵbn = RenavamPipe;
    exports.ɵbo = RGValidator;
    exports.ɵbp = RGPipe;
    exports.ɵbq = TelefoneValidator;
    exports.ɵbr = TelefonePipe;
    exports.ɵbs = TIMEValidator;
    exports.ɵbt = TIMEPipe;
    exports.ɵbu = TITULOValidator;
    exports.ɵbv = TITULOPipe;
    exports.ɵc = TextMaskModule;
    exports.ɵd = cpf;
    exports.ɵe = cnpj;
    exports.ɵf = celular;
    exports.ɵg = cep;
    exports.ɵh = currency;
    exports.ɵi = inscricaoestadual;
    exports.ɵj = number;
    exports.ɵk = pispasep;
    exports.ɵl = placa;
    exports.ɵm = percentage;
    exports.ɵn = renavam;
    exports.ɵo = rg;
    exports.ɵp = telefone;
    exports.ɵq = time;
    exports.ɵr = titulo;
    exports.ɵs = CelularValidator;
    exports.ɵt = CelularPipe;
    exports.ɵu = CEPValidator;
    exports.ɵv = CEPPipe;
    exports.ɵw = CPFValidator;
    exports.ɵx = CPFPipe;
    exports.ɵy = CNPJValidator;
    exports.ɵz = CNPJPipe;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
