// ng-brazil/esm2015/cpf/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { cpf } from './validator';
import * as ɵngcc0 from '@angular/core';
const CPF_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => CPFValidator),
    multi: true
};
export class CPFValidator {
    validate(c) {
        return cpf(c);
    }
}
CPFValidator.ɵfac = function CPFValidator_Factory(t) { return new (t || CPFValidator)(); };
CPFValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CPFValidator, selectors: [["", "cpf", "", "formControlName", ""], ["", "cpf", "", "formControl", ""], ["", "cpf", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CPF_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CPFValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[cpf][formControlName],[cpf][formControl],[cpf][ngModel]',
                providers: [CPF_VALIDATOR]
            }]
    }], null, null); })();
