// ng-brazil/esm2015/currency/directive.js
import { Directive, forwardRef } from '@angular/core';
import { NG_VALIDATORS } from '@angular/forms';
import { currency } from './validator';
import * as ɵngcc0 from '@angular/core';
const CURRENCY_VALIDATOR = {
    provide: NG_VALIDATORS,
    // tslint:disable-next-line: no-use-before-declare
    useExisting: forwardRef(() => CURRENCYValidator),
    multi: true
};
export class CURRENCYValidator {
    validate(c) {
        return currency(c);
    }
}
CURRENCYValidator.ɵfac = function CURRENCYValidator_Factory(t) { return new (t || CURRENCYValidator)(); };
CURRENCYValidator.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: CURRENCYValidator, selectors: [["", "currency", "", "formControlName", ""], ["", "currency", "", "formControl", ""], ["", "currency", "", "ngModel", ""]], features: [ɵngcc0.ɵɵProvidersFeature([CURRENCY_VALIDATOR])] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CURRENCYValidator, [{
        type: Directive,
        args: [{
                // tslint:disable-next-line:directive-selector
                selector: '[currency][formControlName],[currency][formControl],[currency][ngModel]',
                providers: [CURRENCY_VALIDATOR]
            }]
    }], null, null); })();
